### PR TITLE
Adds forgotten Finite Strain class and test for classical plane strain. 

### LINKED ISSUE #5142: Migrate Axisymmetric material from Solid Mechanics to Tensor Mechanics
The start of migrating materials + associated kernels from Solid Mechanics to Tensor Mechanics.  This issue focuses on the axisymmetricRZ "strain calculator" in Solid Mechanics with plans to incorporate the capability through the ComputeStrainBase material in Tensor Mechanics.  Under discussion/ advisement with @jasondhales @bwspenc 

Assign to me

### PR DESCRIPTION
**Pull request by sapitts**
Didn't get this material added in to the last pull request which closed #5142. (so this commit closes #5142 more? )

Includes classical plane strain (uz = 0) for completeness in the migration of the axisymmetric strain calculators.  This material and its small strain analog will be useful as a companion to the future addition of the generalized plane strain code.

### CODE REVIEW

**Comment by jwpeterson** [comment]
Looks like a pretty small diff:
```
2D_geometries.finite_plainstrain:   --------- Time step 1, 0.0000000e+00 ~ 0.0000000e+00, rel diff:  0.00000e+00 ---------
2D_geometries.finite_plainstrain: Nodal variables:
2D_geometries.finite_plainstrain: Element variables:
2D_geometries.finite_plainstrain:   --------- Time step 2, 1.0000000e+00 ~ 1.0000000e+00, rel diff:  0.00000e+00 ---------
2D_geometries.finite_plainstrain: Nodal variables:
2D_geometries.finite_plainstrain: Element variables:
2D_geometries.finite_plainstrain:   --------- Time step 3, 2.0000000e+00 ~ 2.0000000e+00, rel diff:  0.00000e+00 ---------
2D_geometries.finite_plainstrain: Nodal variables:
2D_geometries.finite_plainstrain: Element variables:
2D_geometries.finite_plainstrain:    stress_xx      rel diff:  2.3702569e-09 ~  2.6155591e-09 = 9.37858e-02 (block 0, elmt 2)
2D_geometries.finite_plainstrain:    stress_xy      rel diff:  4.1635942e-10 ~  5.3316733e-10 = 2.19083e-01 (block 0, elmt 3)
```

You might be able to fix it by twiddling the tolerances a bit?  Also, this is a really dumb question but I always thought it was "plane" strain, not "plain"?

File: modules/tensor_mechanics/tests/2D_geometries/tests [hunk 1]
```diff
@@ -4,6 +4,11 @@
     input = 'planestrain_test.i'
     exodiff = 'planestrain_test_out.e'
   [../]
+  [./finite_plainstrain]
```
Comment (dschwen) [inline modules/tensor_mechanics/tests/2D_geometries/tests]: s/plain/plane/g

**Comment by dschwen** [comment]
It is **plane** strain, which is consistent with the class names, but needs to be fixed in the tests file.

**Comment by jwpeterson** [comment]
> Everyone knows this was first observed in the design of aircrafts!

It's plain to see your plain plane planing over the plains.

**Comment by sapitts** [comment]
Yes, of course, I'll fix this spelling error when I fix the exodiff error--just didn't seem prudent to push again solely to fix the spelling.

**Comment by sapitts** [comment]
Tolerances tweaked and spelling changed from boring strain to flat strain.

**Pull request merged**